packages/typespec-azure-core/lib/models.tsp

```diff
@@ -400,3 +400,9 @@ model ArmResourceIdentifierAllowedResource {
    */
   scopes?: ArmResourceDeploymentScope[];
 }
+
+@doc("Used in file part of multipart request body")
+model MultiPartFile extends File {
+  filename: string;
+  contentType: string;
```

Comment:
context: https://github.com/Azure/typespec-azure/issues/1380#issue-2467620574

Reply:
Why contentType is also required? Are user **required** to fill-in the contentType?

Reply:
`Should we have e.g. contentType: "application/octet-stream", if this must be a required property?`
-> It is OK for Python since the default content-type in RFC is also `application/octect-stream`. @chunyu3 / @MaryGao / @qiaozha what's your opinion?

Reply:
Java probably do not want to handle this `contentType: string` as Python.

Making it "required" here basically deprive service dev from actually saying that "contentType" must be filled by user.
If Java add a client default value of "application/octect-stream" in this case of `contentType: string`, how does service dev write in TypeSpec, to have Java not providing this client default value?

Reply:
Java would typically want 

> this contentType must to be sent on wire, but if user does not set anything, SDK would use "application/octet-stream"

I assume this cannot be expressed in TypeSpec without client default value.

But the point, is that I didn't see why the `contentType` property be different from the definition in `TypeSpec.Http.File`. If `contentType` be required here, it should be required in `TypeSpec.Http.File` too.

Reply:
So Java hopes the core model only change `filename` to required.

Reply:
this model is explicitly for a file that has more restrictions on it than a normal `File`. It extends from `File` but overrides the properties to be required instead of optional

Reply:
We could add 2 models for customer:
- `FileWithRequiredMetadata`: both filename and contentType are required (already in this PR)
- `FileWithRequiredFileName`: only filename is required

Then azure service team could choose any one which meets their requirements.

Reply:
I'd assume Java azure-core would only add one class.

loop @srnagar for opinion as well.

Reply:
@srnagar What's your opinion about https://github.com/Azure/typespec-azure/pull/1458#discussion_r1753022700?